autoarray/structures/arrays/kernel_2d.py

```diff
@@ -481,33 +485,90 @@ def convolved_array_from(self, array: Array2D) -> Array2D:
+        jax_method
+            If JAX is enabled this keyword will indicate what method is used for the PSF
+            convolution. Can be either `direct` to calculate it in real space or `fft`
+            to calculated it via a fast Fourier transform. `fft` is typically faster for
+            kernels that are more than about 5x5. Default is `fft`.
+        """
 
-        Returns
-        -------
-        convolved_image
-            An array representing the image after convolution.
+        slim_to_native = jnp.nonzero(
+            jnp.logical_not(image.mask.array), size=image.shape[0]
+        )
+        slim_to_native_blurring = jnp.nonzero(
+            jnp.logical_not(blurring_image.mask.array), size=blurring_image.shape[0]
+        )
 
-        Raises
-        ------
-        KernelException if either Kernel2D psf dimension is odd
+        expanded_array_native = jnp.zeros(image.mask.shape)
+
+        expanded_array_native = expanded_array_native.at[slim_to_native].set(
+            image.array
+        )
+        expanded_array_native = expanded_array_native.at[slim_to_native_blurring].set(
+            blurring_image.array
+        )
+
+        kernel = np.array(self.native.array)
+
+        convolve_native = jax.scipy.signal.convolve(
+            expanded_array_native, kernel, mode="same", method=jax_method
+        )
+
+        convolved_array_1d = convolve_native[slim_to_native]
+
+        return Array2D(values=convolved_array_1d, mask=image.mask)
+
+    def convolve_image_no_blurring(self, image, mask, jax_method="fft"):
         """
+        For a given 1D array and blurring array, convolve the two using this psf.
 
-        if self.mask.shape[0] % 2 == 0 or self.mask.shape[1] % 2 == 0:
-            raise exc.KernelException("Kernel2D Kernel2D must be odd")
+        Parameters
+        ----------
+        image
+            1D array of the values which are to be blurred with the psf's PSF.
+        blurring_image
+            1D array of the blurring values which blur into the array after PSF convolution.
+        jax_method
+            If JAX is enabled this keyword will indicate what method is used for the PSF
+            convolution. Can be either `direct` to calculate it in real space or `fft`
+            to calculated it via a fast Fourier transform. `fft` is typically faster for
+            kernels that are more than about 5x5. Default is `fft`.
+        """
 
-        convolved_array_2d = scipy.signal.convolve2d(array, self.native, mode="same")
+        slim_to_native = jnp.nonzero(jnp.logical_not(mask.array), size=image.shape[0])
 
-        convolved_array_1d = array_2d_util.array_2d_slim_from(
-            mask_2d=np.array(mask),
-            array_2d_native=np.array(convolved_array_2d),
+        expanded_array_native = jnp.zeros(mask.shape)
+
+        expanded_array_native = expanded_array_native.at[slim_to_native].set(image)

```

Comment:
As above this can be done with (again, this should be check to make sure it is doing the same thing).

```python
expanded_array_native = jnp.where(mask.array.T, 0, image)
```